src/API/Middlewares/HttpExceptionsMiddleware.ts

```diff
@@ -14,10 +14,13 @@ export const HttpExceptionsMiddleware: Koa.Middleware = async (context: any, nex
       context.status = 404
     } else {
       if (context.logger)
-        context.logger.error({ exception }, 'Error Caught at Middleware - Will Return Internal Server Error')
+        context.logger.error({ exception }, 'Error Caught at Middleware')
 
-      context.body = 'Internal Server Error'
-      context.status = 503
+      context.body = {
+        message: exception.message || 'Internal Server Error',
+      }
+      context.status = exception.status || 503
+      context.app.emit('error', exception, context)
```

Comment:
Added this because its recommended by koa: https://github.com/koajs/koa/wiki/Error-Handling#catching-downstream-errors